File: ima-backend/src/main/java/com/techsolution/ima_backend/services/MessageService.java
package com.techsolution.ima_backend.services;

import com.techsolution.ima_backend.dtos.request.MessageRequest;
import com.techsolution.ima_backend.dtos.response.MessageResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface MessageService {

    MessageResponse createMessage(MessageRequest messageRequest);
    Page<MessageResponse> getAllMessages(Pageable pageable);
    MessageResponse getMessageById(Long messageId);
    MessageResponse updateMessage(Long messageId, MessageRequest messageRequest);
    void deleteMessage(Long messageId);
}

File: ima-frontend-react/src/services/MessageService.js
import securedAxiosInstance from "./SecuredAxiosInstance";

const REST_API_MESSAGES = '/messages';

export const listMessages = (page, size) => 
    securedAxiosInstance.get(`${REST_API_MESSAGES}?page=${page}&size=${size}`);

export const getMessage = (messageId) => 
    securedAxiosInstance.get(`${REST_API_MESSAGES}/${messageId}`);

export const createMessage = (messageRequest) => 
    securedAxiosInstance.post(`${REST_API_MESSAGES}`, messageRequest);

export const updateMessage = (messageId, messageRequest) => 
    securedAxiosInstance.put(`${REST_API_MESSAGES}/${messageId}`, messageRequest);

export const deleteMessage = (messageId) => 
    securedAxiosInstance.delete(`${REST_API_MESSAGES}/${messageId}`);